api/net/Areas/Admin/Controllers/TopicController.cs

```diff
@@ -132,12 +132,17 @@ public IActionResult Update(TopicModel model)
     [HttpDelete("{id}")]
     [Produces(MediaTypeNames.Application.Json)]
     [ProducesResponseType(typeof(TopicModel), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(string), (int)HttpStatusCode.NoContent)]
     [ProducesResponseType(typeof(ErrorResponseModel), (int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Topic" })]
     public IActionResult Delete(TopicModel model)
     {
-        _service.DeleteAndSave((Topic)model);
-        return new JsonResult(model);
+        var result = _service.FindById(model.Id);
```

Comment:
Don't change the behaviour of this endpoint.  It makes it inconsistent.  Change the behaviour in the UI.  Instead of delete it will update topic by disabling it.

Reply:
Updated, thanks